catroid/src/main/java/org/catrobat/catroid/utils/notifications/StatusBarNotificationManager.java

```diff
@@ -244,58 +209,44 @@ public void showUploadRejectedNotification(int id, int statusCode, String server
 						.setAction(ACTION_CANCEL_UPLOAD);
 				actionIntentCancelUpload.putExtra("bundle", bundle);
-				PendingIntent actionPendingIntentCancelUpload = PendingIntent.getService(context, id,
+				PendingIntent actionPendingIntentCancelUpload = PendingIntent.getService(context, NOTIFICATION_PENDING_INTENT_REQUEST_CODE,
 						actionIntentCancelUpload, PendingIntent.FLAG_ONE_SHOT);
-				notificationBuilder.addAction(android.R.drawable.ic_menu_close_clear_cancel,
-						context.getResources().getString(R.string.cancel), actionPendingIntentCancelUpload);
+				builder.addAction(new NotificationCompat.Action(android.R.drawable.ic_menu_close_clear_cancel,
+						context.getResources().getString(R.string.cancel), actionPendingIntentCancelUpload));
 
 				break;
 			case Constants.STATUS_CODE_UPLOAD_MISSING_CHECKSUM:
 			case Constants.STATUS_CODE_UPLOAD_OLD_CATROBAT_LANGUAGE:
 			case Constants.STATUS_CODE_UPLOAD_OLD_CATROBAT_VERSION:
 				Intent actionIntentUpdatePocketCodeVersion = new Intent(context, NotificationActionService.class)
 						.setAction(ACTION_UPDATE_POCKET_CODE_VERSION)
-						.putExtra("notificationId", id);
-				PendingIntent actionPendingIntentUpdatePocketCodeVersion = PendingIntent.getService(context, id,
+						.putExtra("notificationId", NOTIFICATION_PENDING_INTENT_REQUEST_CODE);
+				PendingIntent actionPendingIntentUpdatePocketCodeVersion = PendingIntent.getService(context, NOTIFICATION_PENDING_INTENT_REQUEST_CODE,
 						actionIntentUpdatePocketCodeVersion, PendingIntent.FLAG_ONE_SHOT);
-				notificationBuilder.addAction(R.drawable.ic_launcher,
-						context.getResources().getString(R.string.notification_open_play_store), actionPendingIntentUpdatePocketCodeVersion);
+				builder.addAction(new NotificationCompat.Action(R.drawable.ic_launcher,
+						context.getResources().getString(R.string.notification_open_play_store), actionPendingIntentUpdatePocketCodeVersion));
 				break;
 
 			default:
-				notificationDataMap.put(id, notificationData);
 				Intent openIntent = new Intent(context, MainMenuActivity.class);
 				openIntent.setAction(Intent.ACTION_MAIN).setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
 						.putExtra(EXTRA_PROJECT_NAME, bundle.getString("projectName"));
 
 				PendingIntent pendingIntent = PendingIntent.getActivity(context, notificationId, openIntent,
 						PendingIntent.FLAG_CANCEL_CURRENT);
-				notificationData.setPendingIntent(pendingIntent);
-				showOrUpdateNotification(id, MAXIMUM_PERCENT);
-				return;
+				builder.setContentIntent(pendingIntent);
+				break;
 		}
 
-		notificationBuilder.setContentIntent(notificationData.getPendingIntent());
-
-		notificationDataMap.put(id, notificationData);
-		notificationManager.notify(id, notificationBuilder.build());
-	}
-
-	public void cancelNotification(int id) {
-		notificationDataMap.remove(id);
-		notificationManager.cancel(id);
-	}
-
-	public int getProgressPercent() {
-		return progressPercent;
+		return builder.build();
```

Comment:
if the builder.build() function really does return the progress in percent then I ask myself why the builder function does this. It's not what I would expect from a builder() function to do.

Reply:
the diff is misleading here. the `getProgressPercent()` function has been removed completely